### PR TITLE
Inspect network requests

### PR DESCRIPTION
**Pull request by methodsecuritysj**
Added the ability to inspect network calls to extract routes. Only done when using browser mode.

To test:

```
go run main.go routecapture browser --target https://convex-nextjs-app-router-demo.vercel.app/ --output json --verbose
go run main.go routecapture browser --target https://www.asksam.com/category/produce/17 --output json --verbose
go run main.go routecapture browser --target http://basicwebsite-postgresdb.method.directory/ --output json --verbose
```

This shows:
* rest api from network requests
* rest and graphql from network requests
* html based parsing

### CODE REVIEW

File: internal/routecapture/helpers.go [hunk 1]
```diff
@@ -111,30 +105,124 @@ func mergeBodyParams(params1, params2 []*webscan.BodyParams) []*webscan.BodyPara
...
+		queryParams = append(queryParams, &webscan.QueryParams{
+			Name:          key,
+			ExampleValues: values,
+		})
+	}
+	return queryParams
+}
+
+// Helper to parse body parameters
+func parseBodyParams(postData string) []*webscan.BodyParams {
```
Comment (sean-hacker) [inline internal/routecapture/helpers.go]: Can we log something or include an error if we find something that is not JSON or form encoded? XML is unfortunately still out there and I'd be interested in having some way to understand if we're seeing it

File: internal/routecapture/networkExtractors.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		errors = append(errors, pageErr.Error())
+		return routes, setToListString(urls), errors
+	}
+	log.Debug("Successfully connected to page for network capture")
+
+	// Stabalize DOM
+	page = page.MustWaitDOMStable()
+
+	// Enable network event tracking
+	networkEventErr := proto.NetworkEnable{}.Call(page)
```
Comment (sean-hacker) [inline internal/routecapture/networkExtractors.go:45]: Do we want to enable the network tracking prior to the DOM stabilizing? I would just worry we may miss some requests.

File: internal/routecapture/networkExtractors.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+		errors = append(errors, networkEventErr.Error())
+		return routes, setToListString(urls), errors
+	}
+
+	// Capture network requests of type 'fetch' which are typical of API calls
+	networkEvents := []*proto.NetworkRequestWillBeSent{}
+	waitForNetworkEvents := page.EachEvent(func(e *proto.NetworkRequestWillBeSent) {
+		log.Debug("Captured network event", svc1log.SafeParam("url", e.Request.URL), svc1log.SafeParam("type", e.Type))
+		// Only capture fetch requests
+		if e.Type == proto.NetworkResourceTypeFetch {
```
Comment (sean-hacker) [inline internal/routecapture/networkExtractors.go:57]: What are the other options?

File: internal/routecapture/networkExtractors.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+			networkEvents = append(networkEvents, e)
+		}
+	})
+
+	// Navigate to the page and wait for the page to load
+	page.MustNavigate(target).MustWaitLoad()
```
Comment (sean-hacker) [inline internal/routecapture/networkExtractors.go:63]: So prior we wait for the DOM to stabilize, but here we navigate to it...again?

File: internal/routecapture/helpers.go (hunk 1 above)
Comment (methodsecuritysj) [reply internal/routecapture/helpers.go]: done

File: internal/routecapture/networkExtractors.go (hunk 3 above)
Comment (methodsecuritysj) [reply internal/routecapture/networkExtractors.go:57]: like document, css, img - if you inspect in your dev tools under "type" you can see; I don't think API calls will go anywhere other than fetch (to my research so far)

File: internal/routecapture/networkExtractors.go (hunk 2 above)
Comment (methodsecuritysj) [reply internal/routecapture/networkExtractors.go:45]: i think the dom stabalize thing was redundant per your other comment, removed

File: internal/routecapture/networkExtractors.go (hunk 4 above)
Comment (methodsecuritysj) [reply internal/routecapture/networkExtractors.go:63]: i think the dom stabalize thing was redundant

**Review by sean-hacker — APPROVED** [review]

**Pull request merged**